### PR TITLE
Test activity for gestures manager

### LINKED ISSUE #11406: Test activity around new gestures library implementation
Adding test activity would a good way to showcase some use-cases and tricks around new gestures implementation.
Having such activity helps us track changes and prevent regression as well.

### PR DESCRIPTION
**Pull request by LukasPaczos**
Closes #11406.
This activity can gradually expand as we add more features through [the gestures library](https://github.com/mapbox/mapbox-gestures-android) and `MapGestureDetector`.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -692,94 +693,105 @@
...
+            android:value="pk.eyJ1IjoiYmxzdGFnaW5nIiwiYSI6ImNpdDF3OHpoaTAwMDcyeXA5Y3Z0Nmk2dzEifQ.0IfB7v5Qbm2MGVYt8Kb8fg" />
 
         <!-- Comment out this setting to switch to external storage (and disable internal) in your app -->
         <!-- <meta-data -->
         <!-- android:name="com.mapbox.SetStorageExternal" -->
         <!-- android:value="true" /> -->
     </application>
 
-</manifest>
+</manifest>
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml]: I think we need to sync our checkstyle configuration for the formatter. To remove the unnecessary changes introduced in this file.

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private RecyclerView recyclerView;
+  private GestureAlertsAdapter gestureAlertsAdapter;
+
+  private AndroidGesturesManager gesturesManager;
+
+  private Marker marker = null;
+  private LatLng focalPointLatLng = null;
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: obsolete null initialisation?

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+import com.mapbox.mapboxsdk.testapp.utils.FontCache;
+import com.mapbox.mapboxsdk.testapp.utils.ResourceUtils;
+
+import java.lang.annotation.Retention;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+public class GestureDetectorActivity extends AppCompatActivity {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: could you add some class level javadoc on what this test activity is testing? 

eg. 
```

/**
 * Test activity showcasing the data driven runtime style API.
 */
```

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+    void addAlert(GestureAlert alert) {
+      for (GestureAlert gestureAlert : alerts) {
+        if (gestureAlert.getAlertType() != GestureAlert.TYPE_PROGRESS) {
+          break;
+        }
+
+        if (alert.getAlertType() == GestureAlert.TYPE_PROGRESS && gestureAlert.equals(alert)) {
+          return;
+        }
+      }
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: not actionable but flagging that code for loops as above with inclusion of break/return increases [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity). This makes reading the code more difficult as it makes predicting the outcome harder.

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml (hunk 1 above)
Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/AndroidManifest.xml]: I thought it's a good time as any other to actually fix the formatting of the Manifest file. Let me know if you'd still rather like to adjust the checkstyle configuration.

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java (hunk 4 above)
Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: Interesting, thanks for pointing this out.

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java (hunk 2 above)
Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: Do you think `null` initialization makes it clearer that this value will be nullified in the future to reflect its state?

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: I'm onboard if that is your way of showing that. For me the final keyword is the one that forces and notifies that something won't be set to null. In any case this is more about code style preference than technical reasoning.

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/GestureDetectorActivity.java]: thinking about this, you could also add `@Nullable` annotation to force null checks with lint
